jobbergate-agent/jobbergate_agent/settings.py

```diff
@@ -3,12 +3,26 @@
 from typing import Optional
 
 import buzz
-from pydantic import AnyHttpUrl, BaseSettings, Field, root_validator
-from pydantic.error_wrappers import ValidationError
+from pydantic import AnyHttpUrl, Field, ValidationError, model_validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from typing_extensions import Self
 
 from jobbergate_agent.utils.logging import logger
 
 
+def _get_env_file() -> str | None:
+    """
+    Determine if running in test mode and return the correct path to the .env file if not.
+    """
+    _test_mode = "pytest" in sys.modules
+    env_file: str | None
+    if not _test_mode:
+        env_file = ".env"
+    else:
+        env_file = None
+    return env_file
```

Comment:
Optional suggestion:

```suggestion
     test_mode = "pytest" in sys.modules
     return None if test_mode else ".env"
```

Reply:
It was left like this after I did some debugging. It's clear enough, I think, as it is.